DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs

```diff
@@ -12,19 +19,72 @@
     [Route("TrackingSystem/Delegates/Groups")]
     public class DelegateGroupsController : Controller
     {
+        private const string DelegateGroupsFilterCookieName = "DelegateGroupsFilter";
+        private static readonly DateTimeOffset CookieExpiry = DateTimeOffset.UtcNow.AddDays(30);
+        private readonly ICustomPromptsService customPromptsService;
         private readonly IGroupsDataService groupsDataService;
 
-        public DelegateGroupsController(IGroupsDataService groupsDataService)
+        public DelegateGroupsController(
+            IGroupsDataService groupsDataService,
+            ICustomPromptsService customPromptsService
+        )
         {
             this.groupsDataService = groupsDataService;
+            this.customPromptsService = customPromptsService;
         }
 
-        public IActionResult Index()
+        [Route("{page=1:int}")]
+        public IActionResult Index(
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
         {
+            // Query parameter should take priority over cookie value
+            // We use this method to check for the query parameter rather 
+            // than filterBy != null as filterBy is set to null to clear 
+            // the filter string when javascript is off.
+            if (!Request.Query.ContainsKey(nameof(filterBy)))
+            {
+                filterBy = Request.Cookies[DelegateGroupsFilterCookieName];
+            }

```

Comment:
I wonder whether we could change the behaviour a bit and maybe add a separate endpoint or a param for clearing filters to make this more intuitive?